pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt

```diff
@@ -485,12 +485,13 @@ class KotlinCodeGenerator(
         builder.addProperty(generateProperty(name, property))
       }
 
-      generateCopyMethods(builder)
-
-      builder
-        .addFunction(generateEqualsMethod())
-        .addFunction(generateHashCodeMethod())
-        .addFunction(generateToStringMethod())
+      if (!pClass.isAbstract) {
```

Comment:
From the commit message:

> Note:
> codegen-java does the same for stateless classes, but it also makes them non-instantiable.
> codegen-kotlin generates stateless regular classes far less often than codegen-java.
 
In other words, nobody bothered to make stateless classes non-instantiable because they are a rare edge case in codegen-kotlin. I didn’t want to change this behavior in this PR, and I did want to generate methods for all instantiable classes (consistent with codegen-java).

Reply:
That's actually a bug; empty classes shouldn't be special-cased here. But, agree with keeping that out-of-scope for this PR.

This code should be able to map to Java:

```
class Empty

empty: Empty
```

Reply:
Actually it’s codegen-java that special-cases stateless classes, probably because that’s what it generates for stateless Pkl modules. So another option would be to change codegen-java to generate instantiable classes for stateless Pkl classes.

Reply:
Sorry, comment on the wrong PR.

But yeah--classes with empty bodies should still have public ctors. Today, this means that `class Empty` can't be correctly mapped to Java.